docs(observe): rework assertions docs around Anomaly Detection and feature-availability stages (#17091)
datahub-project/datahub: docs/managed-datahub/observe/anomaly-detection.md

---
description: This page provides an overview of Anomaly Detection for DataHub Cloud assertions
---

import FeatureAvailability from '@site/src/components/FeatureAvailability';

# Anomaly Detection ⚡

<FeatureAvailability saasOnly />

## What is Anomaly Detection?

**Anomaly Detection** is an optional feature that can be enabled on certain DataHub Cloud assertions to replace fixed thresholds with AI-driven thresholds. Instead of you specifying exact bounds (e.g., "row count must stay between 1,000 and 2,000"), the assertion trains on the historical pattern of the underlying metric and alerts when the latest value falls outside what is considered "normal" — accounting for trends, seasonality, and typical variance.

> Anomaly Detection was previously referred to as **Smart Assertions**. The capability is the same; we now treat it as a feature that can be enabled on an assertion, not a separate class of assertion.

Author: ```suggestion
Anomaly Detection was previously referred to as **Smart Assertions**.
```

<p align="left">
  <img width="80%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/shared/smart-assertion-example.png"/>
</p>

## Where can I enable Anomaly Detection?

Anomaly Detection can be enabled on the following assertion types:

| Assertion type                                              | Anomaly Detection stage | Notes                                                                                                                                                                             |
| ----------------------------------------------------------- | ----------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | [Freshness](./freshness-assertions.md#anomaly-detection-)   | **GA**                  | Works for warehouse-based and ingestion-driven signals. When used with the DataHub Operation aspect, it works on any ingested platform (including Postgres, MySQL, Athena, etc.). |

Reviewer: ```suggestion
| [Freshness](./freshness-assertions.md#anomaly-detection-)   | **GA**                  | Works for warehouse-based and ingestion-driven signals. When used with the DataHub Operation aspect, it works on any ingested platform that reports Operations (including Clickhouse, Oracle, Dremio, etc.). |
```

> | [Volume](./volume-assertions.md#anomaly-detection-)         | **GA**                  | Works for warehouse-based and ingestion-driven signals. When used with DataHub Dataset Profile, it works on any ingested platform.                                                |

Reviewer: ```suggestion
| [Volume](./volume-assertions.md#anomaly-detection-)         | **GA**                  | Works for warehouse-based and ingestion-driven signals. When used with DataHub Dataset Profile, it works on any ingested platform that reports row count profiles (including Iceberg, Postgres, MySQL, etc.).                                                |
```

See the [capabilities matrix](./assertions.md) for the full picture of which assertion types support which signal sources on which platforms.

## Time-Series Bucketing

:::info
Time-series bucketing is currently in **Public Beta**. It is available to all DataHub Cloud customers; we welcome feedback as we continue to iterate.
:::

Anomaly Detection can be configured with **time-series bucketing** to evaluate data quality at a day or week granularity. Instead of checking the entire table each time, the assertion partitions rows into time buckets using a timestamp column and evaluates each bucket independently.

This is especially powerful for Anomaly Detection because it enables the model to learn patterns like "Mondays always have higher volume" or "weekend null counts are typically lower", leading to more accurate anomaly detection.

Time-series bucketing is supported for:

- [Volume Anomaly Detection](./volume-assertions.md#time-series-bucketing)
- [Column Metric Anomaly Detection](./column-assertions.md#time-series-bucketing-for-column-metric-assertions)

Bucketing requires an active warehouse query, so it is only available on Snowflake, Redshift, BigQuery, and Databricks.

> ## Backfill Assertion History

Reviewer: ```suggestion
## Backfill Smart Assertion History
```

Author: I don't think this is necessary

1. I want to kill the "smart assertion" distinction. Just call it anomaly detection
2. We are on the anomaly detection page so it should be obvious what this applies to
